simulate_mining.py

```diff
@@ -53,48 +54,36 @@ def configure_logger(log_filename, timestamp, log_level):
     return logger
 
 def build_stakers(logger, options, timestamp):
-    num_stakers = options.num_stakers
-    whales = options.whales
-    whales_stake = options.whales_stake
-    total_staked = options.total_staked
-    distribution = options.distribution
-    coin_ageing = options.coin_ageing
-    mining_eligibility = options.mining_eligibility
+    num_stakers = options.num_commons + options.num_whales
 
+    distribution = options.distribution
+    
     if distribution == "random":
-        stakers = [random.random() for s in range(num_stakers)]
+        commons = [random.random() for s in range(options.num_commons)]
     elif distribution == "uniform":
-        stakers = [total_staked / num_stakers for s in range(num_stakers)]
+        commons = [options.commons_staked / num_stakers for s in range(options.num_commons)]
     elif distribution == "beta":
-        stakers = [random.betavariate(0.5, 0.2) for s in range(num_stakers)]
+        commons = [random.betavariate(0.5, 0.2) for s in range(options.num_commons)]
     elif distribution == "exponential":
-        stakers = [random.expovariate(2) for s in range(num_stakers)]
+        commons = [random.expovariate(2) for s in range(options.num_commons)]
     elif distribution == "gamma":
-        stakers = [random.gammavariate(0.5, 0.2) for s in range(num_stakers)]
+        commons = [random.gammavariate(0.5, 0.2) for s in range(options.num_commons)]
     else:
         print("Unknown staking distribution")
         sys.exit(1)
 
-    stakers = {i: int(s / sum(stakers) * (total_staked * (100 - whales_stake) / 100)) for i, s in enumerate(stakers)}
-    num_whales = whales
-    while whales > 0:
-        stakers[num_stakers - whales] = int(whales_stake / 100 / num_whales * total_staked)
-        whales -= 1
+    stakers = {
```

Comment:
What is the benefit of changing this such that whales are not inclusive with the total amount of stakers?

Reply:
parameter sweeping simulations based on a pre-fixed amount of stakers